### PR TITLE
Updated GCC 10.x to 11.x in CI tests

### PR DESCRIPTION
**Pull request by MonitSharma**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**   Updated GCC 10.x to 11.x in CI tests

**Description of the Change:**  Changed the version of the gcc and g++ from 10 to 11

**Benefits:**   The update brings improved supports for newer C++ features.

**Possible Drawbacks:**

**Related GitHub Issues:**  [Issue](https://github.com/PennyLaneAI/pennylane-lightning/issues/367)

### CODE REVIEW

**Comment by mlxd** [comment]
Hi @MonitSharma thank you for the PR.

Can you update the PR submission guidelines:
```
Context:

Description of the Change:

Benefits:

Possible Drawbacks:

Related GitHub Issues:
```

with the details of this work, and follow the check-boxes to ensure all requirements are met. Thank you.

You can also ignore the results from `Update dev version automatically` and `codecov/project ` as we will handle those internally.

**Comment by MonitSharma** [comment]
### Before submitting

Please complete the following checklist when submitting a PR:

- [ x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ x] Ensure that the test suite passes, by running `make test`.

- [x ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**Updated GCC 10.x to 11.x in CI tests

**Description of the Change:** Changed the version of the gcc and g++ from 10 to 11

**Benefits:** GCC can now auto-vectorize operations performing addition, subtraction, multiplication and the accumulate/subtract variants on complex numbers.

**Possible Drawbacks:**

**Related GitHub Issues:** none

**Comment by mlxd** [comment]
> ### Before submitting
> 
> Please complete the following checklist when submitting a PR:
> 
>     * [x]  All new features must include a unit test.
>       If you've fixed a bug or added code that should be tested, add a test to the
>       [`tests`](../tests) directory!
> 
>     * [x]  All new functions and code must be clearly commented and documented.
>       If you do make documentation changes, make sure that the docs build and
>       render correctly by running `make docs`.
> 
>     * [x]  Ensure that the test suite passes, by running `make test`.
> 
>     * [ ]  Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
>       change, and including a link back to the PR.
> 
>     * [x]  Ensure that code is properly formatted by running `make format`.
> 
> 
> When all the above are checked, delete everything above the dashed line and fill in the pull request template.
> 
> **Context:**Updated GCC 10.x to 11.x in CI tests
> 
> **Description of the Change:** Changed the version of the gcc and g++ from 10 to 11
> 
> **Benefits:** GCC can now auto-vectorize operations performing addition, subtraction, multiplication and the accumulate/subtract variants on complex numbers.
> 
> **Possible Drawbacks:**
> 
> **Related GitHub Issues:** none

Thanks again @MonitSharma . Can you edit this at the top of the PR (https://github.com/PennyLaneAI/pennylane-lightning/pull/370#issue-1399958278). Also, you did not complete the `Add a new entry to the .github/CHANGELOG.md file, summarizing the change, and including a link back to the PR.` entry. Also for `**Related GitHub Issues:** none` can you include the URL of the original issue that inspired this PR. 

For the listed comments: `GCC can now auto-vectorize operations performing addition, subtraction, multiplication and the accumulate/subtract variants on complex numbers.`, can we verify this is the case for the updated compiler? If not, it should be fine to mention the update brings improved supports for newer C++ features.

Once this is done, we can start a review. Thanks!

**Review by mlxd — APPROVED** [review]
Thanks for the effort @MonitSharma 
Looks good to me! Happy to give it a first approval.

I recommend also that you `Add a new entry to the .github/CHANGELOG.md file, summarizing the
change, and including a link back to the PR.`

We will not accept this as a finished PR until that step is complete :)

**Review by AmintorDusko — CHANGES_REQUESTED** [review]
Hi @MonitSharma, nice work thanks.
You just need now to:
`Add a new entry to the .github/CHANGELOG.md file, summarizing the change, and including a link back to the PR.`

**Comment by MonitSharma** [comment]
I opened another PR for adding summary and my name in the CHANGELOG.md file. I am not sure whether that does it?

**Comment by mlxd** [comment]
> I opened another PR for adding summary and my name in the CHANGELOG.md file. I am not sure whether that does it?

You should be able to make an edit to the changelog in this branch, and have it included as part of this PR. If you are using the Github web interface, this is likely a little trickier than cloning the repository from your branch and editing it there.

however, you can use the in-broser VSCode editor by opening your branch at https://github.com/MonitSharma/pennylane-lightning/tree/patch-1 and hitting the `.` key. This should take you to an interface the looks like 
image

You should be able to edit and save the changes in here into the same branch as your current work. Let me know if this helps!

**Comment by MonitSharma** [comment]
Your message was really helpful, I added the details in the CHANGELOG.md file.

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
 * Upgrade Kokkos and Kokkos Kernels to 3.7.00, and improve sparse matrix-vector multiplication performance and memory usage.
 [(#361)](https://github.com/PennyLaneAI/pennylane-lightning/pull/361)
 
+* Updated GCC 10.x to 11.x in CI tests, changed the version of the gcc and g++ from 10 to 11, and the update brings improved supports for newer C++ features.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:21]: Nice work. Just a small suggestion.
```suggestion
* Update gcc and g++ 10.x to 11.x in CI tests. This update brings improved support for newer C++ features.
[(#370)](https://github.com/PennyLaneAI/pennylane-lightning/pull/370)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -26,7 +29,7 @@
 
 This release contains contributions from (in alphabetical order):
 
-Amintor Dusko, Chae-Yeun Park, Shuli Shu
+Amintor Dusko, Chae-Yeun Park,Monit Sharma, Shuli Shu
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:32]:
```suggestion
Amintor Dusko, Chae-Yeun Park, Monit Sharma, Shuli Shu
```

**Review by AmintorDusko — APPROVED** [review]
You addressed my questions and did a good job!
Thank you for that!

**Review by mlxd — COMMENTED** [review]
Manually updating internal CI dev trigger.

**Comment by mlxd** [comment]
Great work @MonitSharma 

We can call this as approved. We will handle the version string update (Update dev version automatically) and merge this work once we update it.

**Comment by MonitSharma** [comment]
Thank You so much @mlxd and @AmintorDusko

**Pull request merged**